services/editor/src/pages/settings/components/PoliciesPage/utils.ts

```diff
@@ -0,0 +1,64 @@
+
+import * as R from "ramda";
+import {useState, useEffect, useContext} from 'react';
+import { ReduxContext } from "../../../../store";
+import { showError } from "../../../../store/ducks/notifications";
+import { FetchError } from "tweek-client";
+
+type LoadingState = "idle" | "saving" |"loading" | "error";
+
+export function useErrorNotifier(error:Error | null = null, title = "An error has occurred"){
+    const {dispatch} = useContext(ReduxContext);
+    useEffect(()=>{
+        if (!error) return; 
+
+        const format = (error instanceof FetchError) ? ({response: {status, statusText} }:FetchError)=> `${status}: ${statusText}` : 
+                                                       (x:Error) => x.message;
+
+        dispatch(showError({title, error: error, format }))
+    },[error])
+}
+
+export function useRemoteState<T>(reader: ()=> Promise<T>, writer: (data: T) => Promise<void> )
+:[T|null, (data:T)=>void, {loadingState:LoadingState, isDirty: boolean, save: ()=>void, load: ()=>void, error?: Error }] {
+  const [localData,setLocalData] = useState<T|null>(null);
```

Comment:
useReducer? :grimacing:

Reply:
Where?

Reply:
Instead all of the useState

Reply:
It won't make it clearer IMO, the main value I see in reducer is managing complex state that is based on patching previous state. (with the ability to read the previous one)
Don't think it'll better in this case.